.github/workflows/sync-template.yml

```
@@ -0,0 +1,223 @@
+        uses: actions/checkout@v4
+        with:
+          repository: ${{ github.event.inputs.target_repository || inputs.target_repository }}
+          path: 'target-to-synced'
+          token: ${{ steps.get_app_token.outputs.token }}
+
+      - name: Parse configuration
+        id: config
+        run: |
+          CONFIG_FILE="${{ github.event.inputs.config_file || inputs.config_file }}"
+          echo "Using configuration file: $CONFIG_FILE"
+          
+          if [ -f "$CONFIG_FILE" ]; then
+            # Read and parse the configuration file
+            TEMPLATE_REPOSITORY=$(jq -r '.template_repository' "$CONFIG_FILE")
+            WHITELISTED_FILES=$(jq -r '.whitelisted_files | join(" ")' "$CONFIG_FILE")
+            BASE_BRANCH=$(jq -r '.base_branch' "$CONFIG_FILE")
+
+            # Set outputs
+            echo "template_repository=$TEMPLATE_REPOSITORY" >> $GITHUB_OUTPUT
+            echo "base_branch=$BASE_BRANCH" >> $GITHUB_OUTPUT
+            echo "whitelisted_files=$WHITELISTED_FILES" >> $GITHUB_OUTPUT    
+            echo "Configuration loaded successfully"
+          else
+            echo "Warning: Configuration file not found at $CONFIG_FILE"
+            echo "Using default values"
+            
+            # Set default values
+            echo "template_repository=ubiquity/ts-template" >> $GITHUB_OUTPUT
+            echo "base_branch=development" >> $GITHUB_OUTPUT
+            echo "whitelisted_files=.github/* .husky/* .gitattributes .gitignore .nvmrc .prettierrc \
+              .yarnrc.yml .cypress.config.ts eslint.config.mjs jest.config.json package.json \
+              tsconfig.json" >> $GITHUB_OUTPUT
+          fi
+
+          TARGET_REPOSITORY=$(basename ${{ github.event.inputs.target_repository || inputs.target_repository }})
+          echo "target_repository=$TARGET_REPOSITORY" >> $GITHUB_OUTPUT
+
+          BRANCH_REF=sync-template/${BASE_BRANCH}
+          echo "branch_ref=$BRANCH_REF" >> $GITHUB_OUTPUT
```

`BASE_BRANCH` is not defined if the config is not found (else block) - actually it should just exit with error if config is not found

---

Indeed, it is better to just throw an error if config is not found